### DISCUSSION TITLE
Unable to run expo web app without nextjs

### DISCUSSION THREAD

**Discussion by rajatkantinandi** [tamagui/tamagui#1122] [General]
### Current Behavior

Created a expo app using expo typescript template.
Then added react-native-web and added the tamagui libraries as instructed here:

https://tamagui.dev/docs/guides/expo#installing-tamagui-in-expo

But, when I run on web I get this error:
```
No process.env.TAMAGUI_TARGET set, please set it to "native" or "web".
```

I tried running it like this: `TAMAGUI_TARGET=web expo start -c -w` and also tried this `TAMAGUI_TARGET=native expo start -c -w` but it does not work.

But running it on iOS or android works perfectly fine.

### Expected Behavior

Expected to run on web without errors.

### Tamagui Version

```markdown
1.22.11
```

### Reproduction

Try the code in this repo: https://github.com/rajatkantinandi/video-embed-player-native

Run locally by running: `TAMAGUI_TARGET=web expo start -c -w`.

### System Info

```markdown
System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1
    Memory: 101.05 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
```

**Comment by natew** [comment]
Moved to discussion, it's not supported by starters yet.

**Reply by rajatkantinandi** [reply]
Thanks. What did you mean by starters? If expo + react-native-web is not supported then any plan to support it in the near future?

**Comment by ahmouse15** [comment]
Possible workaround: try setting `process.env.TAMAGUI_TARGET = "web";` in the top of your main `App.tsx`

Also see https://github.com/tamagui/tamagui/issues/524#issuecomment-1667010206

**Reply by rajatkantinandi** [reply]
Doesn't seem to work for me. It is still broken.
![Screenshot 2023-08-07 at 10 20 30 AM](https://github.com/tamagui/tamagui/assets/17743895/3297b770-f294-4c0c-aca1-1e920bb8e46e)

**Reply by ahmouse15** [reply]
@rajatkantinandi try putting it before all your imports

**Reply by rajatkantinandi** [reply]
Sorry, still does not seem to work. It is logging it correctly as 'web' from 'App.tsx' but then complaining about it in createComponent.tsx which is strange.

![Screenshot 2023-08-07 at 1 00 35 PM](https://github.com/tamagui/tamagui/assets/17743895/67f889e7-5b87-455f-b6aa-4566e27816d4)

**Reply by rajatkantinandi** [reply]
Does it work with the metro bundler? 
I am currently using Expo SDK 48 with webpack (@expo/webpack-config@18.0.1) and React Native 0.71.7

**Reply by rajatkantinandi** [reply]
Finally able to run it with metro bundler. No need to put `process.env.TAMAGUI_TARGET = "web";` on App.tsx but while running on web we can simply run: `TAMAGUI_TARGET=web expo start -c -w`

**Reply by rajatkantinandi** [reply]
There are some stying issues and layout issues on the web compared to native but that is a separate issue altogether. Glad that it finally works.

**Comment by rajatkantinandi** [comment]
This seems to work with the new metro bundler for react-native and expo SDK 48. So, closing this discussion.
Thanks @ahmouse15 for your suggestion which made me realize that it might work with the metro bundler.
